packages/vow/src/watch.js

```diff
@@ -156,19 +145,11 @@ export const prepareWatch = (
     /** @type {import('./types.js').VowKit<TResult1 | TResult2>} */
     const { resolver, vow } = makeVowKit();
 
-    const { promiseWatcher, vowSetter } = makeWatcherKit(resolver, watcher);
+    // Create a promise watcher to track vows, retrying on disconnection.
+    const promiseWatcher = makePromiseWatcher(resolver, watcher);
 
-    // Ensure we have a presence that won't be disconnected later.
-    unwrapPromise(specimenP, (specimen, payload) => {
-      vowSetter.setVow(specimen);
-      // Persistently watch the specimen.
-      if (!payload) {
-        // Specimen is not a vow.
-        promiseWatcher.onFulfilled(specimen);
-        return;
-      }
-      watchVow(specimen, promiseWatcher);
-    }).catch(e => promiseWatcher.onRejected(e));
+    // Coerce the specimen to a promise, and start the watcher cycle.
+    watchPromise(basicE.resolve(specimenP), promiseWatcher);
```

Comment:
Is this coercion appropriate? @warner: What happens to non-exported local promises when their containing vat is restarted?

Reply:
`watchPromise` adds the promises to liveslots's internal tables, so that they have an id that is reconstituted in the next incarnation.

Reply:
Right, but code anything actually resolve them? I guess the point is moot, since `E.resolve(notAPromise)` is born fulfilled.

Reply:
Looking through `agoric-sdk/packages/swingset-liveslots/src/watchedPromises.js`, it constructs a `loadWatchedPromiseTable` function.  ISTR tracing through this whole flow to find that if a local promise was watched, then the vat was restarted, it was managed correctly.  Something in liveslots would end up calling the `watcher.onRejected` methods for the prior incarnation's promises that were no longer reachable.

But that requires @warner or @mhofman to confirm.

Reply:
Looks like your memory of what happens is correct: https://github.com/Agoric/agoric-sdk/blob/18d561f1b95755e58ca691b26938fc249618018d/packages/SwingSet/test/test-promises.js#L219-L292

But as for the mechanics, it seems to be driven by the kernel rather than liveslots: https://github.com/Agoric/agoric-sdk/blob/18d561f1b95755e58ca691b26938fc249618018d/packages/SwingSet/src/kernel/kernel.js#L902-L905

Albeit with liveslots cooperation in the form of exporting all watched promises: https://github.com/Agoric/agoric-sdk/blob/18d561f1b95755e58ca691b26938fc249618018d/packages/swingset-liveslots/src/watchedPromises.js#L223-L226

So basically, there's no such thing as a non-exported local watched promise.